vault/logical_system_activity.go

```diff
@@ -186,8 +188,13 @@ func (b *SystemBackend) handleActivityConfigUpdate(ctx context.Context, req *log
 	{
 		// Parse the enabled setting
 		if enabledRaw, ok := d.GetOk("enabled"); ok {
+			if config.Enabled == "enable" && enabledRaw.(string) == "disable" {
```

Comment:
enabledRaw could also be "default".

I think it would be better to have SetConfig return the warnings list, so we can handle the 3x3 matrix "enable", "disable", "default" correctly all in one place.  The code there already interprets "default" correctly, so all you have to do is check the original and the final value of a.enabled.

Reply:
But then we have to change the method signature and all corresponding unit tests. Also, `SetConfig(ctx, config) (warnings []string)` seems like a strange API, no? What's wrong with just having these checks here?

Reply:
I would like to avoid duplicating the logic to parse the enabled state and apply it correctly.  (On the other hand, the parsing *does* appear three times in the code already...)

It should be fine to check a.enabled before and after SetConfig instead, but we would probably then need a new IsEnabled() method that handles the mutex locking correctly to avoid setting off the race detector.

Reply:
Sorry, I'm a bit lost here. Is the following not correct?

```go
if config.Enabled == "enable" && (enabledStr == "disable" || enabledStr == "default") {
  warnings = append(warnings, "...")
}
```

I'm not doing any additional parsing here. I'm just checking the new value against the current value.

Reply:
It's correct on OSS but not on enterprise. You would need to incorporate activityLogEnabledDefault, or move the check into a function that is similarly set up to have oss and ent versions.  Something like:

```
if ( config.Enabled == "enable" || (config.Enabled == "default" && activityLogEnabledDefault)) && 
   (enabledStr == "disable" || (enabledStr == "default" && !activityLogEnabledDefault) { 
   ...
}
```